fix: add REQUEST_TIMEOUT 408 as retryable code

cojenco opened the pull request:
This adds `http.client.REQUEST_TIMEOUT` (408) to retryable codes.

Per team discussion with the GCS FnD team, [HTTP status code 408's Request Timeout](https://cloud.google.com/storage/docs/json_api/v1/status-codes#408_Request_Timeout) can be handled with exponential backoff and is retryable. An uploadBrokenConnection will cause a 408. 408 is currently included in [retryable codes in python-storage](https://github.com/googleapis/python-storage/blob/a0f05a6a7176e531af1768ffad12415dbef589b2/google/cloud/storage/retry.py#L42), and should also be included in the resumable media package.

Fixes #271

cojenco committed:
fix: add code 408 as retryable code

tseaver reviewed (changes requested):
The change to add `http.client.REQUEST_TIMEOUT` is fine.  However, the changes to the tests are actually unrelated to that change, and unnecessary -- please revert them.

cojenco committed:
revert 2 test changes

cojenco commented:
For `test_success_with_retry`, I've changed to retry all of the `common.RETRYABLE` codes.  Thus we can ensure that all retryable codes are successfully retried in the mocks. Reverted the 2 other unnecessary changes. PTAL, thanks!

tseaver commented on tests/unit/test__helpers.py:
```diff
@@ -222,16 +217,19 @@ def raise_response():
 
-        assert func.call_count == 4
-        assert func.mock_calls == [mock.call()] * 4
+        assert func.call_count == 7
```
Likewise, can we set the counts here (and below) using something like `len(status_codes)` (or adding / subtracting from it)?

cojenco replied:
Good idea! Revised this test and other related tests in the file.

cojenco committed:
revise test_success_with_retry

cojenco committed:
address review comments
